hooks/darkMode.ts

```diff
@@ -0,0 +1,74 @@
+import DarkModeHookObject from '@interfaces/DarkModeHookObject'
+import { useState, useEffect, useCallback } from 'react'
+
+export const useDarkMode = (): DarkModeHookObject => {
+    const [theme, setTheme] = useState<string>('light')
+    const [logo, setLogo] = useState<string>('https://storage.googleapis.com/sourcegraph-assets/learn/logos/sourcegraph-learn.svg')
+    const [sgLogo, setSgLogo] = useState<string>('https://storage.googleapis.com/sourcegraph-assets/learn/logos/sourcegraph-logo.svg')
+    const isDark = theme === 'dark'
+
+    const setCurrentTheme = (theme: string): void => {
+        window.localStorage.setItem('theme', theme)
+        setTheme(theme)
+    }
+
+    const setCurrentLogo = (theme: string): void => {
+        const logo = window.localStorage.getItem('logo')
+        const sgLogo = window.localStorage.getItem('sg-logo')
+        if (theme === 'dark') {
+            if (!logo) {
+                window.localStorage.setItem('logo', '/static/images/sourcegraph-learn-dark.svg')
```

Comment:
I will update all of these and remove the `public/static/images` directory once we have moved these to the bucket.